pkg/repo/fs.go

```diff
@@ -120,6 +125,16 @@ func (fsr *FsRepo) Open() error {
 		}
 	}
 
+	// check if an instanceID exists persisting one if not found.
+	// never fail here as this isn't critical to node start up.
+	if instanceID, err := fsr.ReadInstanceID(); err != nil {
+		log.Trace().Err(err).Msgf("failed to read instanceID")
```

Comment:
same as above for log level. Also if repo doesn't have system_metadata for some reason, this will throw an error, right? Should we attempt to write the instanceID and recreate the metadata file if thats the case? Or we should handle file not found graceully when reading and return blank insanceID instead of error

Reply:
`fsr.Exists()` will fail if the repo doesn't have `system_metadata.yaml` as it reads the repo version from it.
The only case this will return an error is if a user managers to delete `system_metadata.yaml` after the `Exist` method runs but before this method runs.

In general, my goal with the trace logs here is to avoid notifying a user of any issues with an InstanceID as its presents isn't required for bacalhau to function.